webserver/src/main/java/io/kestra/webserver/services/ai/gemini/GeminiAiService.java

```diff
@@ -0,0 +1,362 @@
+                String mostRelevantTypes = ((String) callbackContext.state().get("mostRelevantTypes")).trim();
+                if (POSSIBLE_ERROR_MESSAGES.contains(mostRelevantTypes)) {
+                    callbackContext.state().put("responseForUser", mostRelevantTypes);
+                    return shortCircuitLlmResponse(callbackContext, mostRelevantTypes);
+                }
+
+                if (mostRelevantTypes.isEmpty()) {
+                    return Maybe.empty();
+                }
+
+                String[] identifiedTypes = mostRelevantTypes.split(" ?, ?");
+
+                JsonNode minifiedSchema = JacksonMapper.ofJson().convertValue(jsonSchemaGenerator.schemas(Flow.class, false, Optional.of(Arrays.asList(identifiedTypes))), JsonNode.class);
+                minifySchema(minifiedSchema);
+                try {
+                    PluginRegistry registry = pluginRegistry;
+                    callbackContext.state().put(
+                        "flowSchema",
+                        JacksonMapper.ofJson().writeValueAsString(minifiedSchema)
+                    );
+                } catch (JsonProcessingException e) {
+                    log.error("Failed to serialize flow schema for AI Agent", e);
+                }
+
+                return Maybe.empty();
+            })
+            .build();
+    }
+
+    private static LlmAgent createYamlBuilderAgent(Gemini model) {
+        return LlmAgent.builder()
+            .name("kestra_init_flow_builder_agent")
+            .description("An agent to bootstrap a Kestra Flow YAML.")
+            .model(model)
+            .generateContentConfig(GenerateContentConfig.builder().temperature(0.9f).maxOutputTokens(50000).build())
+            .outputKey("responseForUser")
+            .beforeAgentCallback((beforeAgentCallback) -> {
+                Object responseForUser = beforeAgentCallback.state().get("responseForUser");
+                if (responseForUser != null) {
+                    return shortCircuitLlmResponse(beforeAgentCallback, responseForUser.toString());
+                }
+
+                return Maybe.empty();
+            })
+            .instruction(
+                """
+                    You are part of a Kestra Flow builder service. You are an expert in generating Kestra Flow YAML. Your task is to generate a valid Kestra Flow YAML based strictly on the following schema:
+                    ```
+                    {{flowSchema}}
+                    ```
+                    
+                    Whenever we're speaking about examples, properties or outputs in the below rules, they are strictly based on the above schema, any external source for those must be discarded because they are not a source of truth.
+                    Here are the rules:
+                    - If the user current flow seems unrelated, you can discard it and start from scratch, otherwise try to keep what you can from the current Flow while still replying to the user's intent.
```

Comment:
```suggestion
                    Guidelines:
                    - Use examples, properties, and outputs only as specified in the schema. 
                    - If the user’s current Flow is unrelated to their prompt, start from scratch; otherwise, retain as much as possible while addressing the user's intent.
```
Worth reiterating to use examples, properties, and outputs only as specified in the schema in the top guideline as it's the main one to follow.